Not compatible with Uppy v4 (could not resolve "@uppy/utils/lib/settle")

Uppy v4.0.0 was released this month (July 2024):
https://github.com/transloadit/uppy/blob/main/CHANGELOG.md#400

Trying to use it with `uppy-activestorage-upload` returns this error:

```
✘ [ERROR] Could not resolve "@uppy/utils/lib/settle"

    node_modules/@excid3/uppy-activestorage-upload/lib/index.js:8:19:
      8 │ import settle from "@uppy/utils/lib/settle"
        ╵                    ~~~~~~~~~~~~~~~~~~~~~~~~

  The path "./lib/settle" is not exported by package "@uppy/utils":

    node_modules/@uppy/utils/package.json:19:13:
      19 │   "exports": {
         ╵              ^

  You can mark the path "@uppy/utils/lib/settle" as external to exclude it from the bundle, which
  will remove this error and leave the unresolved path in the bundle.

1 error
node:child_process:929
    throw err;
```

`settle` was removed here:
https://github.com/transloadit/uppy/pull/5210/files

But is still used here:
https://github.com/excid3/uppy-activestorage-upload/blob/f7b9ed92f1550efa58665341b6423ada0619bcf3/lib/index.js#L165

I think it can be replaced with [`await Promise.allSettled`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled)

--- a/lib/index.js
+++ b/lib/index.js
@@ -5,7 +5,6 @@ import { createId } from "@paralleldrive/cuid2"
 // const { Provider, Socket } = require('@uppy/companion-client');
 // const emitSocketProgress = require('@uppy/utils/lib/emitSocketProgress');
 // const getSocketHost = require('@uppy/utils/lib/getSocketHost');
-import settle from "@uppy/utils/lib/settle"
 import "@uppy/utils/lib/RateLimitedQueue"
 const { DirectUpload } = require("@rails/activestorage")
 
@@ -162,7 +161,7 @@ module.exports = class ActiveStorageUpload extends BasePlugin {
       return limitedAction()
     })
 
-    return settle(promises)
+    return Promise.allSettled(promises)
   }
 
   // Helper to abort upload requests if there has not been any progress for `timeout` ms.

